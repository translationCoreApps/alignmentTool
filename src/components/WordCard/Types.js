export const SECONDARY_WORD = 'bottomWord';
export const PRIMARY_WORD = 'topWord';
