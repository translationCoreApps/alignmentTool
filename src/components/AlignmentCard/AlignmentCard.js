import React, { Component } from 'react';
import PropTypes from 'prop-types';

/**
 * Generates the styles for the component
 * @param props
 * @return {*}
 */
const makeStyles = (props) => {
  const { topWords, hoverTop, hoverBottom, bottomWords, acceptsTopWords, acceptsBottomWords } = props;
  const emptyTop = !topWords || topWords.length === 0;
  const emptyBottom = !bottomWords || bottomWords.length === 0;
  const emptyAlignment = emptyTop && emptyBottom;
  const largeAlignment = (!emptyTop && topWords.length > 1) || (!emptyBottom && bottomWords.length > 1);

  const defaultAlignmentWidth = '115px';
  const blueBorder = '3px dashed #44C6FF';
  const clearBorder = '3px dashed transparent';
  const whiteBorder = '3px dashed #ffffff';
  const transitionSpeed = '0.25s';

  const rowStyle = {
    display: 'flex',
    transition: transitionSpeed,
    position: 'relative'
  };
  const styles = {
    root: {
      transition: transitionSpeed,
      padding: '7px',
      backgroundColor: '#DCDCDC',
      margin: '0px 10px 10px 0px',
      minWidth: emptyAlignment ? `calc(${defaultAlignmentWidth}/2)` : defaultAlignmentWidth,
      flexGrow: largeAlignment ? 1 : 0
    },
    content: {
      display: 'flex',
      flexDirection: 'column',
      height: '100%',
      backgroundColor: '#DCDCDC'
    },
    top: {
      transition: transitionSpeed,
      flexGrow: 1,
      width: '100%',
      minHeight: '45px',
      border: emptyTop || acceptsTopWords ? whiteBorder : clearBorder,
      boxSizing: 'border-box',
      marginBottom: '7px'
    },
    bottom: {
      transition: transitionSpeed,
      flexGrow: 1,
      width: '100%',
      minHeight: '45px',
      border: emptyBottom || acceptsBottomWords ? whiteBorder : clearBorder,
      boxSizing: 'border-box'
    },
    topRow: {
      ...rowStyle,
      top: acceptsTopWords ? '7px' : 0,
      left: acceptsTopWords ? '7px' : 0,
      opacity: hoverTop ? '0.8' : 1
    },
    bottomRow: {
      ...rowStyle,
      top: acceptsBottomWords ? '7px' : 0,
      left: acceptsBottomWords ? '7px' : 0,
      opacity: hoverBottom ? '0.8' : 1
    }
  };

  if (hoverTop && acceptsTopWords) {
    styles.top.border = blueBorder;
  }
  if (hoverBottom && acceptsBottomWords) {
    styles.bottom.border = blueBorder;
  }
  return styles;
};

/**
 * Renders the alignment of primary and secondary words/phrases
 *
 * @property {array} topWords
 * @property {array} bottomWords
 * @property {bool} hoverBottom - a bottom word is hover over this component
 * @property {bool} hoverTop - a top word is hovering over this component
 * @property {bool} acceptsTopWords - this component accepts dropped top words
 * @property {bool} acceptsBottomWords - this component accepts dropped bottom words
 */
class AlignmentCard extends Component {
  render() {
    const { bottomWords, topWords } = this.props;
    const styles = makeStyles(this.props);

    return (
      <div style={styles.root}>
        <div style={styles.content}>
          <div style={styles.top}>
            <div style={styles.topRow}>
              {topWords}
            </div>
          </div>
          <div style={styles.bottom}>
            <div style={styles.bottomRow}>
              {bottomWords}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

AlignmentCard.propTypes = {
  topWords: PropTypes.array,
  bottomWords: PropTypes.array,
  hoverBottom: PropTypes.bool,
  hoverTop: PropTypes.bool,
  acceptsTopWords: PropTypes.bool,
  acceptsBottomWords: PropTypes.bool,
};

export default AlignmentCard;
