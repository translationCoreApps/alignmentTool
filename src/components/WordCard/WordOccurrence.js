import React from 'react';
import PropTypes from 'prop-types';

const styles = {
  top: 7,
  opacity: '0.8',
  marginTop: '5px',
  marginRight: '10px',
  padding: '2px 0px 0px 0px',
};

/**
 * Renders a words occurrence.
 * This is rendered as a superscript.
 * @param {int} occurrence - the order in which this word occurs (1 indexed).
 * @param {int} occurrences - how many times this word occurs in the context
 * @param {object} [style] - overrides the default styles
 * @return {*}
 * @constructor
 */
const WordOccurrence = ({occurrence, occurrences, style}) => {
  const computedStyles = {
    ...styles,
    ...style
  };
  if (occurrences > 1) {
    return <sup style={computedStyles}>{occurrence}</sup>;
  } else {
    return null;
  }
};

WordOccurrence.propTypes = {
  occurrence: PropTypes.number.isRequired,
  occurrences: PropTypes.number.isRequired,
  style: PropTypes.object
};

export default WordOccurrence;
