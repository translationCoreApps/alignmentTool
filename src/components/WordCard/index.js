import React from 'react';
import PropTypes from 'prop-types';
import WordOccurrence from './WordOccurrence';

/**
 * Generates the component styles
 * @param props
 * @return {object}
 */
const makeStyles = (props) => {
  const {onClick, disabled, style, isSuggestion} = props;

  const styles = {
    root: {
      borderLeft: '5px solid #44C6FF',
      padding: '10px',
      backgroundColor: '#FFFFFF',
      boxShadow: '0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset',
      cursor: 'pointer',
      display: 'flex',
      flexDirection: 'row',
      ...style
    },
    word: {}
  };

  if(isSuggestion) {
    styles.root.borderLeft = '5px solid #1b7729';
  }

  if(disabled) {
    styles.root = {
      ...styles.root,
      borderLeft: '5px solid #868686',
      opacity: 0.3,
      cursor: 'not-allowed',
      userSelect: 'none'
    };
  }

  if (!disabled && typeof onClick === 'function') {
    styles.word = {
      ...styles.word,
      cursor: 'pointer'
    };
  }

  return styles;
};

/**
 * Renders a standard word.
 *
 * @param {string} word - the represented word
 * @param {int} occurrence
 * @param {int} occurrences
 * @param {object} [style] - styles passed  through to the component
 * @param {func} [onClick] - callback when the word is clicked
 * @param {bool} [disabled] - indicates the word is disabled
 * @constructor
 */
class WordCard extends React.Component {

  constructor(props) {
    super(props);
    this._handleClick = this._handleClick.bind(this);
  }

  /**
   * Handles click events on the word title.
   * This may block the event if the word is disabled
   * @param e
   * @private
   */
  _handleClick(e) {
    const {disabled, onClick} = this.props;
    if(!disabled && typeof onClick === 'function') {
      onClick(e);
    }
  }

  render() {
    const {word, occurrence, occurrences} = this.props;
    const styles = makeStyles(this.props);
    return (
      <div style={{flex: 1}}>
        <div style={styles.root}>
        <span style={{flex: 1}}>
          <span onClick={this._handleClick} style={styles.word}>
            {word}
          </span>
        </span>
          <WordOccurrence occurrence={occurrence}
                          occurrences={occurrences}/>
        </div>
      </div>
    );
  }
}

WordCard.propTypes = {
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  style: PropTypes.object,
  occurrence: PropTypes.number,
  occurrences: PropTypes.number,
  word: PropTypes.string.isRequired,
  isSuggestion: PropTypes.bool
};

WordCard.defaultProps = {
  style: {},
  occurrence: 1,
  occurrences: 1,
  disabled: false,
  isSuggestion: false
};

export default WordCard;
