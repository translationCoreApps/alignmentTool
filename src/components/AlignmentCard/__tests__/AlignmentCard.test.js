import React from 'react';
import renderer from 'react-test-renderer';
import TestBackend from 'react-dnd-test-backend';
import {DragDropContext} from 'react-dnd';
import AlignmentCard from '../AlignmentCard';
import WordCard from '../../WordCard';

const singleTopWord = [
  <WordCard key={1} word="topWord" />,
];

const singleBottomWord = [
  <WordCard key={1} word="bottomWord" occurrence={1} occurrences={2}  />
];

const multipleTopWords = [
  <WordCard key={1} word="topWord1" />,
  <WordCard key={2} word="topWord2" />
];

const multipleBottomWords = [
  <WordCard key={1} word="bottomWord1" />,
  <WordCard key={2} word="bottomWord2" occurrence={1} occurrences={2} />
];

it('is empty', () => {
  testSnapshot();
});
it('has a top word', () => {
  testSnapshot({
    sourceTokenCards: singleTopWord
  });
});
it('has a top and bottom word', () => {
  testSnapshot({
    sourceTokenCards: singleTopWord,
    targetTokenCards: singleBottomWord
  });
});
it('has a bottom word', () => {
  testSnapshot({
    targetTokenCards: singleBottomWord
  });
});
it('has multiple top words', () => {
  testSnapshot({
    sourceTokenCards: multipleTopWords
  });
});
it('has multiple top and bottom words', () => {
  testSnapshot({
    sourceTokenCards: multipleTopWords,
    targetTokenCards: multipleBottomWords
  });
});
it('has multiple bottom words', () => {
  testSnapshot({
    targetTokenCards: multipleBottomWords
  });
});

/**
 * Tests the snapshot against some props
 * @param props
 */
function testSnapshot(props={}) {
  const ConnectedAlignmentCard = wrapInTestContext(AlignmentCard);
  const wrapper = renderer.create(
    <ConnectedAlignmentCard onDrop={jest.fn()}
                            lexicons={{}}
                            alignmentIndex={0}
                            targetTokenCards={[]}
                            sourceTokenCards={[]}
                            {...props}/>
  );
  expect(wrapper).toMatchSnapshot();
}

/**
 * Wraps a component into a DragDropContext that uses the TestBackend.
 */
function wrapInTestContext(DecoratedComponent) {
  return DragDropContext(TestBackend)(
    class TestContextContainer extends React.Component {
      render() {
        return <DecoratedComponent {...this.props} />;
      }
    }
  );
}
