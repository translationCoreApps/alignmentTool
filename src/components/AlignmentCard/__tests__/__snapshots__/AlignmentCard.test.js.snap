// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has a bottom word 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#DCDCDC",
      "flexGrow": 0,
      "margin": "0px 10px 10px 0px",
      "minWidth": "115px",
      "padding": "7px",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "border": "3px dashed #ffffff",
          "boxSizing": "border-box",
          "flexGrow": 1,
          "marginBottom": "7px",
          "minHeight": "45px",
          "transition": "0.1s",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "left": 0,
            "opacity": 1,
            "position": "relative",
            "top": 0,
            "transition": "0.1s",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "border": "3px dashed transparent",
          "boxSizing": "border-box",
          "direction": "ltr",
          "flexGrow": 1,
          "minHeight": "45px",
          "transition": "0.1s",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "left": 0,
            "opacity": 1,
            "position": "relative",
            "top": 0,
            "transition": "0.1s",
          }
        }
      >
        <div
          aria-describedby={null}
          className=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
          title={null}
        >
          <div
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderLeft": "5px solid #44C6FF",
                "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                "cursor": "pointer",
                "display": "flex",
                "flexDirection": "row",
                "padding": "9px",
              }
            }
          >
            <span
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <span
                className={undefined}
                onClick={[Function]}
                style={
                  Object {
                    "flexGrow": 2,
                    "overflow": "hidden",
                    "textOverflow": "ellipsis",
                    "width": "max-content",
                  }
                }
              >
                bottomWord
              </span>
            </span>
            <sup
              style={
                Object {
                  "opacity": "0.8",
                  "padding": "2px 0px 0px",
                  "top": "0px",
                }
              }
            >
              1
            </sup>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`has a top and bottom word 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#DCDCDC",
      "flexGrow": 0,
      "margin": "0px 10px 10px 0px",
      "minWidth": "115px",
      "padding": "7px",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "border": "3px dashed transparent",
          "boxSizing": "border-box",
          "flexGrow": 1,
          "marginBottom": "7px",
          "minHeight": "45px",
          "transition": "0.1s",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "left": 0,
            "opacity": 1,
            "position": "relative",
            "top": 0,
            "transition": "0.1s",
          }
        }
      >
        <div
          aria-describedby={null}
          className=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
          title={null}
        >
          <div
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderLeft": "5px solid #44C6FF",
                "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                "cursor": "pointer",
                "display": "flex",
                "flexDirection": "row",
                "padding": "9px",
              }
            }
          >
            <span
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <span
                className={undefined}
                onClick={[Function]}
                style={
                  Object {
                    "flexGrow": 2,
                    "overflow": "hidden",
                    "textOverflow": "ellipsis",
                    "width": "max-content",
                  }
                }
              >
                topWord
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "border": "3px dashed transparent",
          "boxSizing": "border-box",
          "direction": "ltr",
          "flexGrow": 1,
          "minHeight": "45px",
          "transition": "0.1s",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "left": 0,
            "opacity": 1,
            "position": "relative",
            "top": 0,
            "transition": "0.1s",
          }
        }
      >
        <div
          aria-describedby={null}
          className=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
          title={null}
        >
          <div
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderLeft": "5px solid #44C6FF",
                "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                "cursor": "pointer",
                "display": "flex",
                "flexDirection": "row",
                "padding": "9px",
              }
            }
          >
            <span
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <span
                className={undefined}
                onClick={[Function]}
                style={
                  Object {
                    "flexGrow": 2,
                    "overflow": "hidden",
                    "textOverflow": "ellipsis",
                    "width": "max-content",
                  }
                }
              >
                bottomWord
              </span>
            </span>
            <sup
              style={
                Object {
                  "opacity": "0.8",
                  "padding": "2px 0px 0px",
                  "top": "0px",
                }
              }
            >
              1
            </sup>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`has a top word 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#DCDCDC",
      "flexGrow": 0,
      "margin": "0px 10px 10px 0px",
      "minWidth": "115px",
      "padding": "7px",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "border": "3px dashed transparent",
          "boxSizing": "border-box",
          "flexGrow": 1,
          "marginBottom": "7px",
          "minHeight": "45px",
          "transition": "0.1s",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "left": 0,
            "opacity": 1,
            "position": "relative",
            "top": 0,
            "transition": "0.1s",
          }
        }
      >
        <div
          aria-describedby={null}
          className=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
          title={null}
        >
          <div
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderLeft": "5px solid #44C6FF",
                "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                "cursor": "pointer",
                "display": "flex",
                "flexDirection": "row",
                "padding": "9px",
              }
            }
          >
            <span
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <span
                className={undefined}
                onClick={[Function]}
                style={
                  Object {
                    "flexGrow": 2,
                    "overflow": "hidden",
                    "textOverflow": "ellipsis",
                    "width": "max-content",
                  }
                }
              >
                topWord
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "border": "3px dashed #ffffff",
          "boxSizing": "border-box",
          "direction": "ltr",
          "flexGrow": 1,
          "minHeight": "45px",
          "transition": "0.1s",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "left": 0,
            "opacity": 1,
            "position": "relative",
            "top": 0,
            "transition": "0.1s",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`has multiple bottom words 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#DCDCDC",
      "flexGrow": 1,
      "margin": "0px 10px 10px 0px",
      "minWidth": "115px",
      "padding": "7px",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "border": "3px dashed #ffffff",
          "boxSizing": "border-box",
          "flexGrow": 1,
          "marginBottom": "7px",
          "minHeight": "45px",
          "transition": "0.1s",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "left": 0,
            "opacity": 1,
            "position": "relative",
            "top": 0,
            "transition": "0.1s",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "border": "3px dashed transparent",
          "boxSizing": "border-box",
          "direction": "ltr",
          "flexGrow": 1,
          "minHeight": "45px",
          "transition": "0.1s",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "left": 0,
            "opacity": 1,
            "position": "relative",
            "top": 0,
            "transition": "0.1s",
          }
        }
      >
        <div
          aria-describedby={null}
          className=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
          title={null}
        >
          <div
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderLeft": "5px solid #44C6FF",
                "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                "cursor": "pointer",
                "display": "flex",
                "flexDirection": "row",
                "padding": "9px",
              }
            }
          >
            <span
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <span
                className={undefined}
                onClick={[Function]}
                style={
                  Object {
                    "flexGrow": 2,
                    "overflow": "hidden",
                    "textOverflow": "ellipsis",
                    "width": "max-content",
                  }
                }
              >
                bottomWord1
              </span>
            </span>
          </div>
        </div>
        <div
          aria-describedby={null}
          className=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
          title={null}
        >
          <div
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderLeft": "5px solid #44C6FF",
                "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                "cursor": "pointer",
                "display": "flex",
                "flexDirection": "row",
                "padding": "9px",
              }
            }
          >
            <span
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <span
                className={undefined}
                onClick={[Function]}
                style={
                  Object {
                    "flexGrow": 2,
                    "overflow": "hidden",
                    "textOverflow": "ellipsis",
                    "width": "max-content",
                  }
                }
              >
                bottomWord2
              </span>
            </span>
            <sup
              style={
                Object {
                  "opacity": "0.8",
                  "padding": "2px 0px 0px",
                  "top": "0px",
                }
              }
            >
              1
            </sup>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`has multiple top and bottom words 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#DCDCDC",
      "flexGrow": 1,
      "margin": "0px 10px 10px 0px",
      "minWidth": "115px",
      "padding": "7px",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "border": "3px dashed transparent",
          "boxSizing": "border-box",
          "flexGrow": 1,
          "marginBottom": "7px",
          "minHeight": "45px",
          "transition": "0.1s",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "left": 0,
            "opacity": 1,
            "position": "relative",
            "top": 0,
            "transition": "0.1s",
          }
        }
      >
        <div
          aria-describedby={null}
          className=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
          title={null}
        >
          <div
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderLeft": "5px solid #44C6FF",
                "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                "cursor": "pointer",
                "display": "flex",
                "flexDirection": "row",
                "padding": "9px",
              }
            }
          >
            <span
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <span
                className={undefined}
                onClick={[Function]}
                style={
                  Object {
                    "flexGrow": 2,
                    "overflow": "hidden",
                    "textOverflow": "ellipsis",
                    "width": "max-content",
                  }
                }
              >
                topWord1
              </span>
            </span>
          </div>
        </div>
        <div
          aria-describedby={null}
          className=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
          title={null}
        >
          <div
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderLeft": "5px solid #44C6FF",
                "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                "cursor": "pointer",
                "display": "flex",
                "flexDirection": "row",
                "padding": "9px",
              }
            }
          >
            <span
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <span
                className={undefined}
                onClick={[Function]}
                style={
                  Object {
                    "flexGrow": 2,
                    "overflow": "hidden",
                    "textOverflow": "ellipsis",
                    "width": "max-content",
                  }
                }
              >
                topWord2
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "border": "3px dashed transparent",
          "boxSizing": "border-box",
          "direction": "ltr",
          "flexGrow": 1,
          "minHeight": "45px",
          "transition": "0.1s",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "left": 0,
            "opacity": 1,
            "position": "relative",
            "top": 0,
            "transition": "0.1s",
          }
        }
      >
        <div
          aria-describedby={null}
          className=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
          title={null}
        >
          <div
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderLeft": "5px solid #44C6FF",
                "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                "cursor": "pointer",
                "display": "flex",
                "flexDirection": "row",
                "padding": "9px",
              }
            }
          >
            <span
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <span
                className={undefined}
                onClick={[Function]}
                style={
                  Object {
                    "flexGrow": 2,
                    "overflow": "hidden",
                    "textOverflow": "ellipsis",
                    "width": "max-content",
                  }
                }
              >
                bottomWord1
              </span>
            </span>
          </div>
        </div>
        <div
          aria-describedby={null}
          className=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
          title={null}
        >
          <div
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderLeft": "5px solid #44C6FF",
                "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                "cursor": "pointer",
                "display": "flex",
                "flexDirection": "row",
                "padding": "9px",
              }
            }
          >
            <span
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <span
                className={undefined}
                onClick={[Function]}
                style={
                  Object {
                    "flexGrow": 2,
                    "overflow": "hidden",
                    "textOverflow": "ellipsis",
                    "width": "max-content",
                  }
                }
              >
                bottomWord2
              </span>
            </span>
            <sup
              style={
                Object {
                  "opacity": "0.8",
                  "padding": "2px 0px 0px",
                  "top": "0px",
                }
              }
            >
              1
            </sup>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`has multiple top words 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#DCDCDC",
      "flexGrow": 1,
      "margin": "0px 10px 10px 0px",
      "minWidth": "115px",
      "padding": "7px",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "border": "3px dashed transparent",
          "boxSizing": "border-box",
          "flexGrow": 1,
          "marginBottom": "7px",
          "minHeight": "45px",
          "transition": "0.1s",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "left": 0,
            "opacity": 1,
            "position": "relative",
            "top": 0,
            "transition": "0.1s",
          }
        }
      >
        <div
          aria-describedby={null}
          className=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
          title={null}
        >
          <div
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderLeft": "5px solid #44C6FF",
                "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                "cursor": "pointer",
                "display": "flex",
                "flexDirection": "row",
                "padding": "9px",
              }
            }
          >
            <span
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <span
                className={undefined}
                onClick={[Function]}
                style={
                  Object {
                    "flexGrow": 2,
                    "overflow": "hidden",
                    "textOverflow": "ellipsis",
                    "width": "max-content",
                  }
                }
              >
                topWord1
              </span>
            </span>
          </div>
        </div>
        <div
          aria-describedby={null}
          className=""
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
          title={null}
        >
          <div
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderLeft": "5px solid #44C6FF",
                "boxShadow": "0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset",
                "cursor": "pointer",
                "display": "flex",
                "flexDirection": "row",
                "padding": "9px",
              }
            }
          >
            <span
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <span
                className={undefined}
                onClick={[Function]}
                style={
                  Object {
                    "flexGrow": 2,
                    "overflow": "hidden",
                    "textOverflow": "ellipsis",
                    "width": "max-content",
                  }
                }
              >
                topWord2
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "border": "3px dashed #ffffff",
          "boxSizing": "border-box",
          "direction": "ltr",
          "flexGrow": 1,
          "minHeight": "45px",
          "transition": "0.1s",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "left": 0,
            "opacity": 1,
            "position": "relative",
            "top": 0,
            "transition": "0.1s",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`is empty 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#DCDCDC",
      "flexGrow": 0,
      "margin": "0px 10px 10px 0px",
      "minWidth": "calc(115px/2)",
      "padding": "7px",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "border": "3px dashed #ffffff",
          "boxSizing": "border-box",
          "flexGrow": 1,
          "marginBottom": "7px",
          "minHeight": "45px",
          "transition": "0.1s",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "left": 0,
            "opacity": 1,
            "position": "relative",
            "top": 0,
            "transition": "0.1s",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "border": "3px dashed #ffffff",
          "boxSizing": "border-box",
          "direction": "ltr",
          "flexGrow": 1,
          "minHeight": "45px",
          "transition": "0.1s",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "left": 0,
            "opacity": 1,
            "position": "relative",
            "top": 0,
            "transition": "0.1s",
          }
        }
      />
    </div>
  </div>
</div>
`;
